### PR TITLE
Added purchseProduct function

### PR DESCRIPTION
**Pull request by suriksarkisyan**

### CODE REVIEW

File: src/classes/Qonversion.ts [hunk 1]
```diff
@@ -70,8 +70,27 @@ export default class Qonversion {
   }
 
   static async purchase(productId: string): Promise<Map<string, Permission>> {
+    return this.purchaseProxy(productId);
+  }
+
+  /**
+   * Starts a process of purchasing product with Qonversion's [product] object.
+   * @param product - Qonversion's product object
```
Comment (Maria-Bordunova) [inline src/classes/Qonversion.ts:78]:
```suggestion
   * @param product - Qonversion's Product object
```

File: src/classes/Qonversion.ts [hunk 2]
```diff
@@ -70,8 +70,27 @@ export default class Qonversion {
   }
 
   static async purchase(productId: string): Promise<Map<string, Permission>> {
+    return this.purchaseProxy(productId);
+  }
+
+  /**
+   * Starts a process of purchasing product with Qonversion's [product] object.
```
Comment (Maria-Bordunova) [inline src/classes/Qonversion.ts:77]:
```suggestion
   * Starts a process of purchasing product with Qonversion's [Product] object.
```

**Review by Maria-Bordunova — APPROVED** [review]

File: android/src/main/java/com/reactlibrary/QonversionModule.java [hunk 3]
```diff
@@ -120,15 +121,65 @@ public void onError(@NotNull QonversionError qonversionError) {
...
             QonversionError qonversionError = generateActivityError();
             promise.reject(qonversionError.getCode().toString(),qonversionError.getDescription());
             return;
         }
 
+        if (offeringId != null) {
+            Qonversion.offerings(new QonversionOfferingsCallback() {
+                @Override
+                public void onSuccess(@NotNull QOfferings offerings) {
+                    QOffering offering = offerings.offeringForID(offeringId);
```
Comment (SpertsyanKM) [inline android/src/main/java/com/reactlibrary/QonversionModule.java:146]: I'd offer to mark all immutable variables as `final`

File: android/src/main/java/com/reactlibrary/QonversionModule.java [hunk 4]
```diff
@@ -120,15 +121,65 @@ public void onError(@NotNull QonversionError qonversionError) {
...
+                                @Override
+                                public void onSuccess(@NotNull Map<String, QPermission> map) {
+                                    WritableMap result = EntitiesConverter.mapPermissions(map);
+                                    promise.resolve(result);
+                                }
+
+                                @Override
+                                public void onError(@NotNull QonversionError qonversionError) {
+                                    rejectWithError(qonversionError, promise);
+                                }
```
Comment (SpertsyanKM) [inline android/src/main/java/com/reactlibrary/QonversionModule.java:162]: If we put it into variable, we'd simplify the code

**Review by SpertsyanKM — DISMISSED** [review]

**Review by SpertsyanKM — APPROVED** [review]

**Pull request merged**